backend/local/backend_refresh_test.go

```diff
@@ -135,10 +135,6 @@ func TestLocal_refreshValidate(t *testing.T) {
 	}
 	<-run.Done()
 
```

Comment:
To the concerns about impacting other operations: You could add another test here where the provider is configured, and ensure that this call does indeed happen in that test. Optional, but if you're interested in more programmatically exploring some of the concerns you raised.